functions/src/motherboard/parseMotherboardPage.ts

```diff
@@ -0,0 +1,106 @@
+import { Page } from 'puppeteer';
+import { Motherboard, MotherboardFormFactor } from '../../../types';
+import camelize from '../common/camelize';
+import cleanComplexTable from '../common/cleanComplexTable';
+import { xPathSelectors } from '../common/constants';
+import getParsingElement from '../common/getParsingElement';
+import parseElementInnerHTML from '../common/parseElementInnerHTML';
+import parseElementText from '../common/parseElementText';
+
+const parseMotherboardPage = async (
+  productId: string,
+  page: Page,
+): Promise<Motherboard | null> => {
+  const description = await parseElementInnerHTML('.desc-ai-title', page);
+
+  await page.waitForXPath(xPathSelectors.specificationButton);
+  const anchor = (await page.$x(xPathSelectors.specificationButton)) as any;
+  await Promise.all([
+    await anchor[0].click(),
+    await page.waitForNavigation({ waitUntil: 'networkidle2' }),
+  ]);
+
+  const name = await parseElementText('.op1-tt', page);
+
+  const mainImageContainer = await getParsingElement('.img200', page);
+  const mainImage = await page.evaluate(
+    (el) => el.lastElementChild.getAttribute('srcset').split(' ')[0],
+    mainImageContainer,
+  );
+
+  const specsTable = await getParsingElement('#help_table', page);
+
+  const rawSpecsTable = await page.evaluate(async (node) => {
+    async function getNodeTreeText(inputNode: Element): Promise<string | null> {
+      if (inputNode && inputNode.hasChildNodes()) {
+        return node.innerText;
+      }
+
+      return null;
+    }
+
+    return getNodeTreeText(node);
+  }, specsTable);
+
+  if (!name || !mainImage || !rawSpecsTable) return null;
+
+  const cleanedSpecsTable = cleanComplexTable(rawSpecsTable);
+
+  const specs: Record<string, string> = {};
+
+  cleanedSpecsTable.forEach((item: string) => {
+    const [name, value] = item.split('\t');
+
+    if (!name && !value) {
+      return;
+    }
+
+    const camelName = camelize(name);
+    specs.hasOwnProperty(camelName)
+      ? (specs[`${camelName}Internal`] = value)
+      : (specs[camelName] = value);
+  });
+
+  console.log(specs);
+
+  return {
+    id: productId,
+    name,
+    mainImage,
+    description: description || undefined,
+    socket: specs?.socket,
+    formFactor: specs?.formFactor as MotherboardFormFactor,
+    powerPhases: specs?.powerPhases,
+    VRMHeatsink: !specs?.vRMHeatsink,
+    size: specs?.['size(HxW)'], // e.g. 226x211 mm
+    chipset: specs?.chipset,
+    BIOS: specs?.BIOS,
+    DDR4: specs?.DDR4,
+    memoryModule: specs?.memoryModule,
+    operationMode: specs?.operationMode,
+    maxClockFrequency: specs?.maxClockFrequency,
+    maxMemory: specs?.maxMemory,
+    VGA: !specs?.['dSubOutput(VGA)'],
+    HDMI: !specs?.hDMIOutput,
+    HDMIVersion: specs?.hDMIVersion,
+    displayPort: !specs?.displayPort,

```

Comment:
why ! if it is boolean use !!

Reply:
it either does not exist, or has a value of "". ! is the simplest way to put true here